File: internal/indexer/testdata/js/services/criticalUserModelService.js
const { CriticalUserModel } = require("../models/criticalUserModel");

class CriticalUserModelRepository {
  constructor() {
    this.store = [];
  }

  save(model) {
    this.store.push(model);
    return true;
  }

  findByEmail(email) {
    return this.store.find((m) => m.email === email) || null;
  }

  findAll() {
    return this.store;
  }
}

function storeCriticalUserModel(model) {
  return model instanceof CriticalUserModel && model.validate();
}

module.exports = { CriticalUserModelRepository, storeCriticalUserModel };

File: internal/indexer/testdata/python/services/critical_user_model_service.py
from models.critical_user_model import CriticalUserModel


class CriticalUserModelRepository:
    """Manages CriticalUserModel persistence."""

    def __init__(self):
        self._store: list = []

    def save(self, model: CriticalUserModel) -> bool:
        self._store.append(model)
        return True

    def find_by_email(self, email: str):
        for m in self._store:
            if isinstance(m, CriticalUserModel) and m.email == email:
                return m
        return None

    def find_all(self) -> list:
        return list(self._store)


def store_critical_user_model(model: CriticalUserModel) -> bool:
    """Standalone function for persisting a CriticalUserModel."""
    return model.validate()
